### DISCUSSION TITLE
Prevent response from "re-validating" before being returned via API

### DISCUSSION THREAD

**Discussion by robswc** [vitalik/django-ninja#1012] [Q&A]
Hey! Have been using django-ninja for awhile and love what it's been able to do for our org!

One small problem I keep running into is when I provide context, via the `model_validate` method, to create the models, it works perfectly fine.  However, when I return the model, it gets "re-validated" before being returned via the API, and it essentially wipes the custom context that the resolvers rely on.

```
things =  [
    Thing.model_validate(model_to_dict(t),
                                          context={
                                              'custom': 'stuff',
                                          }) for t in things
]
```

Is there a way to approach this problem?

Thanks!

**Comment by robswc** [comment]
At the moment, I just have:

```
response={
    200: List[Any]}
```

As a quick and dirty fix, but it removes the openapi schema info.

**Reply by jannekanerva** [reply]
Thanks, this worked for me also. Have you found any other way of preventing the re-validation?

**Reply by robswc** [reply]
I don't know if it is a good method, but I was able to override:
```
_run_root_validator = custom_validator
```

With a custom validator, for the schema that I needed to avoid double-validation on.

```
@model_validator(mode="wrap")
def custom_validator(cls, values: Any, handler: ModelWrapValidatorHandler[S], info: ValidationInfo) -> Any:

    if info.context:
       # do stuff with context here
       
    forbids_extra = cls.model_config.get("extra") == "forbid"
    should_validate_assignment = cls.model_config.get("validate_assignment", False)
    if forbids_extra or should_validate_assignment:
        handler(values)

    values = DjangoGetter(values, cls, info.context)
    return handler(values)
   ```
